doc/changelog/06-Ltac2-language/16537-ltac2-more-kind-equal.rst

```diff
@@ -0,0 +1,9 @@
+- **Added:**
+  ``Ltac2.Constant.equal``, ``Ltac2.Constant.t``, ``Ltac2.Constructor.equal``,
```

Comment:
Would you add a sentence of description at the beginning?  I'm not sure a long list of items is really that helpful here.  

OTOH, we have roughly zero documentation on stdlib and it's quite hard to find things in it.  Let alone in packages outside stdlib.  We need better indexing and searching tools and (better) doc.  Something to keep in mind, but let's not discuss today.

Reply:
What sort of sentence are you looking for?  I've added a bunch of type aliases `Ltac2.Foo.t` for `foo` when not already present, and a bunch of functions `Ltac2.Foo.equal` for boolean equality between `Ltac2.Foo.t`.  Most of these are types that show up in `Ltac2.Constr.Unsafe.kind`, which is the primary motivation for adding them.  If a user reading the changelog doesn't already know about the term AST, then I'm not sure what would be helpful here; if the user already knows the structure of Coq's term AST, then the functionality is obvious from the names (and where it's not obvious, that's because of design questions that we're picking more-or-less arbitrary answers to, such as "should `Case.equal` ignore universes?")

Reply:
There are no universes in case_info
This PR has no questions related to universes

Reply:
unless you count relevances I suppose

Reply:
Maybe say something like

```
 **Added:**
  [Selected?] Missing type aliases `Ltac2.Foo.t` for `foo`, and equality functions `Ltac2.Foo.equal`.
```

But also make clear whether these are syntactic items (since you mention the AST) or whatever they are (I don't know).

Reply:
The current changelog looks good enough to me.